text/0000-cargo-libstd-awareness.md

```diff
@@ -0,0 +1,236 @@
+If and when Rust 2, a version with breaking changes, comes out, this will be all the more important.
+We don't yet have a plan yet to track the language itself, but by tracking standard library dependencies we make it trivial to specify version requirements like any other package.
+
+
+# Detailed design
+
+## Standard library dependencies
+
+First and foremost, one will now be able to explicitly depend on standard library crates, e.g. with `std = { version = "1.10", stdlib = true }`.
+From the users's perspective, `stdlib = true` simply indicates that the depended-on crate is from the standard library.
+The version for stdlib crates comes from the version of Rust their interfaces are defined in.
+A version requirement must be specified.
+The full breadth of options available with our existing dependencies, e.g. features and overrides, will be supported.
+
+## Implicit dependencies
+
+For backwards compatibility, Cargo must inject such standard library dependencies for existing packages.
+These injected standard library dependencies are called "implicit dependencies" because the user does not specify them explicitly.
+We have an obligation to not break packages depending only on stable interfaces, so the implicit dependencies will include both `std` and `core`:
+```toml
+[dependencies]
+core = { version = "^1.0", stdlib = true }
+std  = { version = "^1.0", stdlib = true }
+
+[dev-dependencies]
+core = { version = "^1.0", stdlib = true }
+std  = { version = "^1.0", stdlib = true }
+test = { version = "^1.0", stdlib = true }
+
+[build-dependencies]
+core = { version = "^1.0", stdlib = true }
+std  = { version = "^1.0", stdlib = true }
+```
+The version requirement for `core` of `^1.0` may seem odd because core was not stable at the time, but anything else would either break newer packages using core, or prevent older packages from working on versions of Rust predating core's stabilization.
+Remember that rustc only complains if an unstable crate is actually imported, so the Cargo dependency on its own is harmless.
+
+`test` is a similar scenario.
+While importing it explicitly remains unstable, it's currently injected and thus needs to be built.
+Other dependencies of `std` besides core we don't need to worry about, because they are only transitive dependencies through `std`, not direct dependencies.
+
+Now, not all crates depend on these crates, so there must be a way to opt out.
+For this, we introduce a new `implicit-dependencies` key.
+It is defined by default as:
+```toml
+implicit-dependencies = ["primary", "build", "dev"]
+```
+This indicates each of `dependencies`, `build-dependencies`, and `dev-dependencies` maps (respectively) is augmented with implicit elements.
+A manual definition may be that or almost any subset, in which case only the included dependency maps are augmented.
+
+Finally, if an (explicit) dependency conflicts with one of the implicit defaults, that category of implicit dependency will be skipped.
```

Comment:
Instead of "category" -> "the implicit dependencies for that compilation profile will be omitted"?